cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1084,19 +1104,25 @@ func NewBicepProvider(
 		return nil, err
 	}
 
+	projectConfig, err := project.LoadProjectConfig(filepath.Join(projectPath, azdcontext.ProjectFileName))
```

Comment:
If we would rather say that `*project.ProjectConfig` is something that flows across the Provider interface (and hence would be passed as an argument to the constructor here by the provisioning manager), I'm happy to make that change.  I thought about it for a little bit (and indeed I think that the `infraOptions Options` is actually a sub part of that larger object) but wasn't 100% sure about it, so I opted for this change which reused the existing project path to just reload the file, since that was less disruptive to the rest of the code.  Wanted to get a read from y'all on what you think the right thing is to do here.

Reply:
Are we using the `projectConfig` for any other than `to.Ptr(p.projectConfig.Name),` ? 

Should we save `NewBicepProvider` from creating all the `projectConfig` if it only needs the project name?

I would like to avoid adding the relation from package `bicep` to package `project` 
```
import "github.com/azure/azure-dev/cli/azd/pkg/project"
```
if the bicep is only expecting to know a project name

Reply:
I would vote for:

1. Rename `provisioning.Options` to `provisioning.Config`, since it really is a configuration stored inside project config for the infra provisioning.
2. Make the current `options` an actual "options" for constructing the manager, i.e. struct values, not dependencies.
3. Put stuff outside of `provisioning.Config`, but part of `projectConfig` inside the options. This includes the `projectName`.

Alternatively, we could also have `provisioning.Config` reference `project.Config` similar to how we do in other places, although I've always thought that ends up leaking too much.

Reply:
One of the motivations for keeping the project/service config out of the provisioning provider was to keep it slim and only need the `infra` config/options on the project/service being deployed.  

If we are able to keep it this way and only expose the infra related options that would allow us to keep a more loosely coupled design.